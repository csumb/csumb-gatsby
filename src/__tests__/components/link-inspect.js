import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import LinkInspect from 'components/utilities/link-inspect'

describe('LinkInspect', () => {
  it('renders a base relative URL correctly', () => {
    const tree = renderer
      .create(<LinkInspect to="/location">A link</LinkInspect>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an absolute relative URL correctly', () => {
    const tree = renderer
      .create(<LinkInspect to="https://google.com">A link</LinkInspect>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders nothing if there is no "to" prop', () => {
    const tree = renderer.create(<LinkInspect>A link</LinkInspect>).toJSON()
    expect(tree).toBeNull()
  })
})
