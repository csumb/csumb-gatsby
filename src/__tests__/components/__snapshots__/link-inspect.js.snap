// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkInspect renders correctly 1`] = `
<a
  href="/location"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  A link
</a>
`;
